### PR TITLE
Update SDK Version Prefix

### PR DESCRIPTION
**Pull request by jeremydvoss**
# Description

Update SDK Version Prefix according to [spec](https://github.com/aep-health-and-standards/Telemetry-Collection-Spec/blob/main/ApplicationInsights/sdk_version_name.md#allowed-values-for-the-attach-type-character)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Release History
 
+## 1.0.0b23 (Unreleased)
```
Comment (lzchen) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:3]: After releases, please make sure to appr and merge the automatically created post-rlease prs. Then you wont have to do these manually. https://github.com/Azure/azure-sdk-for-python/pull/34198

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_version.py [hunk 2]
```diff
@@ -5,4 +5,4 @@
 # license information.
 # --------------------------------------------------------------------------
 
-VERSION = "1.0.0b22"
+VERSION = "1.0.0b23"
```
Comment (lzchen) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_version.py:8]: Already done by: https://github.com/Azure/azure-sdk-for-python/pull/34198

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_utils.py [hunk 3]
```diff
@@ -36,23 +36,40 @@
 
 
 def _is_on_app_service():
-    return "WEBSITE_SITE_NAME" in environ
+    return environ.get("WEBSITE_SITE_NAME") is not None
 
+def _is_on_function():
```
Comment (lzchen) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_utils.py:41]:
```suggestion
def _is_on_functions():
```

**Review by lzchen — APPROVED** [review]

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md (hunk 1 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:3]: That [task failed](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3471076&view=logs&j=9e408172-77a7-548b-8cd1-e45278a8c989&t=e65141db-c30e-5b82-93af-081ba0717a0b), so I assumed the PR wasn't created. Strange.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_utils.py (hunk 3 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_utils.py:41]: Done.

**Comment by jeremydvoss** [comment]
Converting to draft until p360 mapping is agreed upon by all language leads

**Pull request merged**